### DISCUSSION TITLE
Need help interpreting results MiXCR TCR-seq

### DISCUSSION THREAD

**Discussion by JoostKoedijk** [milaboratory/mixcr#1797] [Q&A] [answered]
Hi, I am in need of some help with the interpretation of my TCR-seq results. My aim is to assess the clonality / diversity of TCR clones at the beta subunit at diagnosis, after the first course of chemotherapy and after the second course of chemotherapy. I have performed standard bulk RNA-sequencing with about 70-100 million reads and now applied MiXCR (v. 4.6.0) and the postanalysis option. 

I figured out that I would like to calculate the TCR CPK metric (clonotypes per kilobase reads) for the beta subunit. For that I would need to have the unique number of clonotypes for each sample and divide that by the number of reads for the TRB chain, right? 

Here is the alignment report, which I think I should use to find the number of reads for the TRB chain: 821 reads. Correct?

Analysis time: 371.12m
Total sequencing reads: 37593498
Successfully aligned reads: 138261 (0.37%)
Coverage (percent of successfully aligned):
  CDR3: 66872 (48.37%)
  FR3_TO_FR4: 7 (0.01%)
  CDR2_TO_FR4: 11 (0.01%)
  FR2_TO_FR4: 10 (0.01%)
  CDR1_TO_FR4: 6 (0%)
  VDJRegion: 6 (0%)
Alignment failed: no hits (not TCR/IG?): 36551690 (97.23%)
Alignment failed: no CDR3 parts: 155791 (0.41%)
Alignment failed: low total score: 747756 (1.99%)
Overlapped: 0 (0%)
Overlapped and aligned: 0 (0%)
Overlapped and not aligned: 0 (0%)
Alignment-aided overlaps, percent of overlapped and aligned: 0 (NaN%)
No CDR3 parts alignments, percent of successfully aligned: 1097 (0.79%)
Partial aligned reads, percent of successfully aligned: 70292 (50.84%)
Realigned with forced non-floating bound: 0 (0%)
Realigned with forced non-floating right bound in left read: 0 (0%)
Realigned with forced non-floating left bound in right read: 0 (0%)
TRA chains: 574 (0.42%)
TRA non-functional: 40 (6.97%)
TRB chains: 821 (0.59%)
TRB non-functional: 15 (1.83%)
TRD chains: 61 (0.04%)
TRD non-functional: 0 (0%)
TRG chains: 245 (0.18%)
TRG non-functional: 17 (6.94%)
IGH chains: 58557 (42.35%)
IGH non-functional: 204 (0.35%)
TRAD chains: 97 (0.07%)
TRAD non-functional: 0 (0%)
IGK chains: 47410 (34.29%)
IGK non-functional: 158 (0.33%)
IGL chains: 30496 (22.06%)
IGL non-functional: 85 (0.28%)
Trimming report:
  R1 reads trimmed left: 57441 (0.15%)
  R1 reads trimmed right: 133395 (0.35%)
  Average R1 nucleotides trimmed left: 0.010745528388978328
  Average R1 nucleotides trimmed right: 0.11893370497206725
======================================

Then, to find the number of unique clonotypes, I have a look at the CDR3 beta subunit spectratyping results (see attached excel file). 
[CDR3 spectratyping beta subunit MiXCR results.xlsx](https://github.com/user-attachments/files/17045905/CDR3.spectratyping.beta.subunit.MiXCR.results.xlsx). 
When counting the number of unique clones, I would like to take all the the clones which have the number null after the number of nucleotides (e.g., 45 null), as I read in another post that the clones with their aa sequence written in full are part of the ones which have null after the number of nucleotides. However, then I don't know whether these reads are unique or clonal, right? Can you explain how I find the number of unique clones in this case? 

Many, many thanks!!

**Comment by mizraelson** [comment, marked as answer]
Yes. The total number of reads aligned to TRB is 821 (functional) + 15 (non-functional) = 836
The number of unique clonotypes corresponds to the number of rows in the output clonotype table for the TRB. This table lists each distinct clonotype identified in your sample. Alternatively, you can find this information in the assembly report, which provides a summary of the assembly statistics, including the total number of unique clonotypes detected per chain.

**Reply by JoostKoedijk** [reply]
OK, thank you very much!! 

To be sure about the unique clonotypes: sometimes there are rows (clones) that are not present at diagnosis (DX) but present at end of induction 1 or 2 (EOI1/2). For instance, for clone '28 null' at DX (Diagnosis) and EOI2 (end of induction 2), the clone abundance is null, so I guess I shouldn't count this as a unique clone for DX and EOI2, correct? Only for EOI1?
![image](https://github.com/user-attachments/assets/981bc674-782d-43bd-b5ae-e88601c0844a)

And do you mean that those rows where the amount of nucleotides is followed by null (e.g., '36 null') is one unique clone? And not the sum of all clones with nucleotide length 36? Perhaps to clarify, how would you count the number of unique clones for DX, EOI1, and EOI2 in the attached excel sheet (including how you got to each number)? 

I really appreciate your prompt answers, thank you! Very kind.

**Reply by JoostKoedijk** [reply]
So for instance, if column DX is filtered by all but null, I get to 19 rows with a number behind it. Is that the number of unique clones? Does that mean that the clone '45 null' is present in 11 T cells?
![image](https://github.com/user-attachments/assets/c043ca85-5aaa-46f4-b814-5aedfdb2fb42)

**Reply by mizraelson** [reply]
You need the output clonotype tsv tables from the analyze command, not postanalysis.

**Reply by JoostKoedijk** [reply]
One more question on the clonotype output: 

(Let's assume that these clones are also similar for the D and J segment) -- Are TRBV13*00(455.8) and TRBV13*00(502.9) the same clones? (in other words, what does the number between brackets indicate)?

**Reply by mizraelson** [reply]
The number in the brackets is the alignment score. TRBV13*00 (455.8) and TRBV13*00 (502.9) mean that these clones use the same V gene. Even when clones have the same V, D, and J genes, they can be different because they might have different CDR3 region sequence due to added nucleotides. In the output clonotype table, no records have identical nucleotide sequences.

**Reply by JoostKoedijk** [reply]
Ok :) so to see if a clone is identical, I should check whether the V, D and J genes AND the aa sequence are similar between samples?

**Reply by mizraelson** [reply]
It depends whether you are interested in AA or nucleotide sequences. But generally speaking, yes.

You can also use the [following command](https://mixcr.com/mixcr/reference/mixcr-exportClonesOverlap/) to see identical clones between multiple samples:
```
mixcr exportClonesOverlap --criteria "CDR3|AA|V|J"  *.clns overlapTable.tsv 
```
